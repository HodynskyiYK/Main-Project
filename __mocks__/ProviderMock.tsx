import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}