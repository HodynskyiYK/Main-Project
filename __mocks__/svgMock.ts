// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock