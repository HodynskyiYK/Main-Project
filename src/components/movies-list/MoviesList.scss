.movies-counter {
  padding: 1.25rem 0;

  p {
    color: $white_color;
    font: 400 1.25rem/1.2em $main_font;
    text-transform: lowercase;

    b {
      font-weight: 600;
    }
  }
}

.movies-list {
  padding-bottom: 1.5rem;
}

.movie-card {
  margin-bottom: 3.125rem;

  .card-header {
    position: relative;
    margin-bottom: 1.25rem;
  }

  .movie-link {
    position: relative;
    display: block;
    font: inherit;
    z-index: 1;
  }

  .movie-image {
    display: block;
  }

  .edit-btn {
    position: absolute;
    top: 1rem;
    right: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 2.75rem;
    height: 2.75rem;
    border-radius: 2.25rem;
    background: #2A202D;
    box-shadow: 0 0 4px rgba(0, 0, 0, 0.196596);
    overflow: hidden;
    padding: 0;
    border: 0;
    opacity: 0;
    cursor: pointer;
    z-index: 3;

    svg {
      display: block;
    }
  }

  .card-body {
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    justify-content: flex-start;
  }

  .movie-title {
    color: rgba($white_color, 0.7);
    font: 500 1.125rem/1.2em $main_font;
    flex: 0 0 calc(100% - 4.375rem);
    max-width: calc(100% - 4.375rem);
    margin-bottom: 0.375rem;

    a {
      position: relative;
      display: block;
      font: inherit;
      z-index: 1;
    }
  }

  .release {
    flex: 0 0 4.375rem;
    max-width: 4.375rem;
    color: rgba($white_color, 0.7);
    font: 500 0.875rem/1.2em $main_font;
    text-align: center;
    border: 1px solid #979797;
    border-radius: 4px;
    padding: 0.375rem 0;
    margin-bottom: 0.375rem;
  }

  .genre {
    flex: 0 0 100%;
    max-width: 100%;
    color: rgba($white_color, 0.5);
    font: 500 0.875rem/1.2em $main_font;
  }

  &:hover {

    .edit-btn {
      opacity: 1;
    }
  }
}