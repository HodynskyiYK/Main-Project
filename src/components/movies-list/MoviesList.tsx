import React, {FC} from 'react'
import {IMoviesList} from './moviesTypes'
import {MovieCard} from './MovieCard'

export const MoviesList: FC<IMoviesList> = ({moviesList}) => {

    return (
        <>
            <div className="movies-counter">
                <p>
                    <b>39</b> movies found
                </p>
            </div>
            <div className="movies-list row">{
                moviesList.map((movieItem) => (
                    <MovieCard key={movieItem.id} movieItem={movieItem} />
                ))
            }</div>
        </>
    )
}
