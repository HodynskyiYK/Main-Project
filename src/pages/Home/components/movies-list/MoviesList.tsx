import React, { FC, useEffect } from 'react'
import classnames from 'classnames'
import {MovieCard} from '../movie-card'
import { useMovieModalContext } from '../../../../context/movie-modal-context'
import { MovieModalForm } from '../movie-modal-form'
import { DeleteMovieModal } from '../delete-movie-modal'
import { MOVIE_INFORMATION } from './moviesListConstans'
import { useMovieDetailsContext } from '../../../../context/movie-details-context'
import { Loader } from '../../../../shared/loader'
import { useAction, useTypedSelector } from '../../../../hooks'
import styles from './MoviesList.module.scss'
import { EmptyMovieList } from './EmptyMovieList'
// import { Button } from '../../../../components/button'

export const MoviesList: FC = () => {
    const {fetchMovies} = useAction()
    const {movies, isLoading, filterBy, sortBy} = useTypedSelector(state => state.movies)
    const {
        editMovieModalState,
        deleteMovieModalState,
        showEditMovieModal,
        showDeleteMovieModal
    } = useMovieModalContext()

    const {updateMovieDetails} = useMovieDetailsContext()

    useEffect(() => {
        fetchMovies(filterBy, sortBy)
    // eslint-disable-next-line
    }, [])

    // console.log(movies)
    return (
        <>
            <div className={styles.moviesCounter}>
                <p>
                    <b>{movies.length}</b> movies found
                </p>
            </div>
            {
                movies.length ? (
                    <div className={classnames('row', styles.moviesList)}>{
                        movies.map((movieItem) => (
                            <MovieCard
                                key={movieItem.id}
                                movieItem={movieItem}
                                editMovie={showEditMovieModal}
                                deleteMovie={showDeleteMovieModal}
                                getMovieDetails={updateMovieDetails}
                            />
                        ))
                    }</div>
                ) : (
                    !isLoading && <EmptyMovieList/>
                )
            }
            {
                isLoading && (
                    <Loader />
                )
            }
            {/*<div className={styles.loadMore}>
                <Button
                    buttonClassName={'addBtn'}
                    buttonText={isLoading ? 'Loading...' : 'Load more'}
                    buttonDisabled={isLoading}
                    buttonAction={() => fetchMovies()}
                />
            </div>*/}
            {

            }
            {
                editMovieModalState && <MovieModalForm modalName={'Edit movie'} formValues={MOVIE_INFORMATION} />
            }
            {
                deleteMovieModalState && <DeleteMovieModal />
            }
        </>
    )
}
