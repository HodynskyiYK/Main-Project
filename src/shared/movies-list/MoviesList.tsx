import React, { FC, useEffect, useState } from 'react'
import classnames from 'classnames'
import {MovieCard} from '../movie-card'
import { useMovieModalContext } from '../../context/movie-modal-context'
import { MovieModalForm } from '../movie-modal-form'
import { DeleteMovieModal } from '../delete-movie-modal'
import { MOVIE_INFORMATION } from './moviesListConstans'
import { useMovieDetailsContext } from '../../context/movie-details-context'
import { Loader } from '../loader'
import { useAction, useTypedSelector } from '../../hooks'
import styles from './MoviesList.module.scss'
import { EmptyMovieList } from './EmptyMovieList'
import { IMovieItem } from '../../store/actions-types'

export const MoviesList: FC = () => {
    const [editMovie, setEditMovie] = useState<IMovieItem>(MOVIE_INFORMATION)
    const [deleteMovieId, setDeleteMovieId] = useState<number>(0)
    const {fetchMovies} = useAction()
    const {movies, isLoading, filterBy, sortBy} = useTypedSelector(state => state.movies)
    const {
        editMovieModalState,
        deleteMovieModalState,
        showEditMovieModal,
        showDeleteMovieModal
    } = useMovieModalContext()

    const {updateMovieDetails} = useMovieDetailsContext()

    useEffect(() => {
        fetchMovies(filterBy, sortBy)
    // eslint-disable-next-line
    }, [])

    const getEditMovie = (movie: IMovieItem) => {
        setEditMovie(movie)
        showEditMovieModal()
    }

    const getEditMovieId = (id: number | undefined) => {
        if (id) {
            setDeleteMovieId(id)
            showDeleteMovieModal()
        }
    }

    return (
        <>
            <div className={styles.moviesCounter}>{
                <p>
                    <b>{movies.length}</b> movies found
                </p>
            }</div>
            {
                movies.length ? (
                    <div className={classnames('row', styles.moviesList)}>{
                        movies.map((movieItem) => (
                            <MovieCard
                                key={movieItem.id}
                                movieItem={movieItem}
                                editMovie={getEditMovie}
                                deleteMovie={getEditMovieId}
                                getMovieDetails={updateMovieDetails}
                            />
                        ))
                    }</div>
                ) : (
                    !isLoading && <EmptyMovieList/>
                )
            }
            {
                isLoading && (
                    <Loader />
                )
            }
            {
                editMovieModalState && <MovieModalForm modalName={'Edit movie'} formValues={editMovie} />
            }
            {
                deleteMovieModalState && <DeleteMovieModal movieId={deleteMovieId} />
            }
        </>
    )
}
