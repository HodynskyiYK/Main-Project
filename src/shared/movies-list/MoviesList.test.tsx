import React from 'react'
import { render } from '@testing-library/react'
import { MoviesList } from './MoviesList'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('MoviesList tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock>
            <MoviesList/>
        </ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})