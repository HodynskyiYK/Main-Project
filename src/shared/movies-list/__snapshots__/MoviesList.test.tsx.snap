// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="moviesCounter"
  >
    <p>
      <b>
        0
      </b>
       movies found
    </p>
  </div>
  <div
    class="loaderContent"
  >
    <span
      class="loader"
    />
  </div>
</DocumentFragment>
`;
